include/vcpkg/archives.h

```diff
@@ -13,16 +13,36 @@
 
 namespace vcpkg
 {
+    enum class ExtractionType
+    {
+        UNKNOWN,
+        TAR,
+        ZIP,
+        NUPKG,
+        MSI,
+        EXE
+    };
+
     // Extract `archive` to `to_path` using `tar_tool`.
     void extract_tar(const Path& tar_tool, const Path& archive, const Path& to_path);
     // Extract `archive` to `to_path` using `cmake_tool`. (CMake's built in tar)
     void extract_tar_cmake(const Path& cmake_tool, const Path& archive, const Path& to_path);
-    // Extract `archive` to `to_path`, deleting `to_path` first.
     void extract_archive(const Filesystem& fs,
                          const ToolCache& tools,
                          MessageSink& status_sink,
                          const Path& archive,
                          const Path& to_path);
+    // set `to_path` to `archive` contents.
+    void set_directory_to_archive_contents(const Filesystem& fs,
```

Comment:
I renamed it to "set_directory..." because the function deletes whatever is in the target directory before extracting, essentially setting the directory to the contents of the archive.

Reply:
What about `set_directory_contents_to_archive_contents`?